src/main/java/com/tradeshift/service/HelloWorldServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.tradeshift.service;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.tradeshift.model.Content;
+import com.tradeshift.model.DBMessage;
+import com.tradeshift.model.Message;
+import com.tradeshift.model.RecentModel;
+
+public class HelloWorldServiceImpl implements HelloWorldService {
+
+    @Autowired
+    private MessagesDAO messagesDAO;
+
+    public HelloWorldServiceImpl() {
+    }
+
+    public HelloWorldServiceImpl(MessagesDAO messagesDAO) {
+        this.messagesDAO = messagesDAO;
+    }
+
+    @Override
+    public String getHelloWorldMessage(String name){
+
+        return "hello " + name;
+    }
+
+    @Override
+    public void insert(String name) {
+        if(name == null && name.isEmpty()) {
+            throw new NullPointerException("String is null or empty");
+        }
+        this.messagesDAO.insert(name);
+    }
+
+    @Override
+    public RecentModel getMessages(int amount) {
+
+        List<DBMessage> retrievedMessages = messagesDAO.getMessages(10);
+
+        if (retrievedMessages.size() > 0) {
+
+            List<Message> messages = new ArrayList<>();
+            for (DBMessage temp : retrievedMessages) {
+                messages.add(new Message(new Content(getHelloWorldMessage(temp.getName()))));
```

Comment:
This could be a candidate for a builder pattern. Not in scope for this problem unless you really want to, but it's a good thing to keep in mind for later when there are similar situations :smile: